### PR TITLE
[js]Update * to originHost

### LINKED ISSUE #19358: JavaScript Security Vulnerability
The issues were found when performing a code scan using SonarQube.  The application found 3 security vulnerabilities.  I am including them in a single submission since it is the same issue found in 3 places in two different files.  Matamo version 4.10.1

File number one:  js/pwiki.js   Line: 7236

iframe.contentWindow.postMessage(jsonMessage, '*');

SonarQube reason: Origins should be verified during cross-origin communications

Our recommendation: create a URL variable that defaults to '*' but allow the value to be defined/overridden in a configuration setting.

File number two: plugins/CoreAdminHome/javascripts/optOut.js  Lines: 105 and 200

Line 105: parent.postMessage(JSON.stringify(optOutStatus), "*");

Line 200: parent.postMessage(JSON.stringify(message), '*');

SonarQube reason: Origins should be verified during cross-origin communications

Our recommendation: create a URL variable that defaults to '*' but allow the value to be defined/overridden in a configuration setting.

If you need additional information, please let me know.

### PR DESCRIPTION
**Pull request by peterhashair**
### Description:
Fixes: #19358 
Only fix apply issue 1 fixes. update * to `originHost`. It seems e.origin contain protocol already.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by peterhashair** [comment]
build js

**Comment by peterhashair** [comment]
build js

**Comment by peterhashair** [comment]
build js

File: js/piwik.js [hunk 1]
```diff
@@ -7427,7 +7427,7 @@ if (typeof window.Matomo !== 'object') {
 
                     if (iframe.contentWindow && isDefined(iframe.contentWindow.postMessage) && iframeHost === originHost) {
                         var jsonMessage = JSON.stringify(postMessage);
-                        iframe.contentWindow.postMessage(jsonMessage, '*');
+                        iframe.contentWindow.postMessage(jsonMessage, originHost);
```
Comment (sgiehl) [inline js/piwik.js]: Guess you didn't have a deeper look at the code. It's correct that `e.origin` contains the protocol, but `originHost` is set to `getHostName(e.origin)`, which removes everything, but the hostname.

**Review by sgiehl — CHANGES_REQUESTED** [review]

**Comment by peterhashair** [comment]
build js

File: js/piwik.js [hunk 2]
```diff
@@ -7465,7 +7465,8 @@ if (typeof window.Matomo !== 'object') {
                     maq_optout_by_default: tracker.isConsentRequired()
                 });
             }
-        }, false);
+        }, fal
+        se);
```
Comment (sgiehl) [inline js/piwik.js]: Guess you added this line break by accident:

```suggestion
        }, false);
```

**Comment by peterhashair** [comment]
build js

**Comment by sgiehl** [comment]
The code changes are looking ok.
@peterhashair what exactly did you do to test the changes locally? I did not yet have a look for what exactly the changed code is used, would be helpful if you could provide some insights, so I don't need to spent too much time to find some code that allows testing it.

**Comment by justinvelluppillai** [comment]
ping @peterhashair this is waiting for your reply to @sgiehl to help him review this effectively.

**Comment by peterhashair** [comment]
@sgiehl @justinvelluppillai ops, sorry, I think there is js test for it, which is that one. https://github.com/matomo-org/matomo/blob/a8dec66081fa5cd671a668ef248c36e9d3285ffc/tests/javascript/index.php#L5219

But to tests locally. I add this to my 
```javascript
     addEventListener('load', (event) => {
                const targetFrame = window.top.frames[0];
                const targetOrigin = 'https://yourtrackingsite';
                targetFrame.postMessage('{"d":"hello there","maq_initial_value":true}', targetOrigin);
            });

```

**Comment by sgiehl** [comment]
@peterhashair the PR contains two added files, that shouldn't exist.

**Comment by peterhashair** [comment]
@sgiehl ops, that was accidentally added, removed.

**Review by sgiehl — APPROVED** [review]

**Pull request merged**